datadict/datadict-sqlite.inc.php

```diff
@@ -84,12 +84,16 @@ function actualType($meta)
 	}
 
 	// return string must begin with space
-	function _createSuffix($fname, &$ftype, $fnotnull, $fdefault, $fautoinc, $fconstraint, $funsigned)
+	function _createSuffix($fname,&$ftype,$fnotnull,$fdefault,$fautoinc,$fconstraint,$funsigned,$fprimary,&$pkey)
 	{
 		$suffix = '';
-		if ($funsigned) $suffix .= ' UNSIGNED';
+		if ($funsigned && !($fprimary && $fautoinc)) $suffix .= ' UNSIGNED';
 		if ($fnotnull) $suffix .= ' NOT NULL';
 		if (strlen($fdefault)) $suffix .= " DEFAULT $fdefault";
+		if ($fprimary && $fautoinc) {
+			$suffix .= ' PRIMARY KEY';
+			array_pop($pkey);
+		}
 		if ($fautoinc) $suffix .= ' AUTOINCREMENT';
```

Comment:
I don't think that's right, per https://www.sqlite.org/syntax/column-constraint.html `AUTOINCREMENT` is not valid on its own, must follow `PRIMARY KEY`.

Guess it should be 
```
if ($fprimary) {
	$suffix .= ' PRIMARY KEY';
	array_pop($pkey);
    if ($fautoinc) $suffix .= ' AUTOINCREMENT';
}
```

Reply:
Good point.

Reply:
It's OK to have a separate PRIMARY definition if AUTOINCREMENT is not specified.